### ISSUE TITLE
There should be an easy way to use fmt_number on all numeric columns

### ISSUE THREAD

**Issue by ymer** [rstudio/gt#613]
fmt_number() should be able to work on a dataframe that has a mix of numeric and other columns, without having to name all the columns to be formatted.

(This may already be possible, but I can't figure out how from the description https://gt.rstudio.com/reference/fmt_number.html )

**Comment by rich-iannone** [comment]
This is great. I'm looking forward to a future in gt where columns can have a mixture of cells with different types (this would be accomplished by having some table composition functions).

I really like the idea of formatting all numbers by default with `fmt_number()`. You can specify the constraints with `columns` and `rows` and non-numeric cells are ignored if encountered. Same goes for things like dates and times (format all that can be formatted). Until the building/structuring of heterogeneous tables is fully thought out, modifications like this to the formatters cannot really proceed though.

**Comment by LukasWallrich** [comment]
Thanks for looking into this, it would be very helpful. Just in case anyone comes across this in the meantime, looking for a workaround - the following worked for me, even though there must be a simpler way to pass the digits argument from gt::fmt?

```
fmt_if_number <- function(..., digits = 2) {
  input <- c(...)
  fmt <- paste0("%.", digits, "f")
  if (is.numeric(input))   return(sprintf(fmt, input))
  return(input)
}

cbind(rownames(mtcars), mtcars[1:2]) %>%
  gt::gt() %>%
  gt::fmt(gt::everything(), fns = function(x) 
    fmt_if_number(x, digits = 1))
```

**Comment by ymer** [comment]
Re-reading this, I'm not sure we are talking about the same thing. Let me try with an example:

`tibble(name = c("Benny", "Jets"), n = c(1.0001, 7.001)) %>% 
   gt() %>% 
   fmt_number(vars(n), decimals = 1)`
   
Here I would like to be able to instead use:

  `fmt_number(gt::numeric(), decimals = 1)`

**Comment by LukasWallrich** [comment]
Good idea - but with that, the issue becomes identical with #618?

**Comment by rich-iannone** [comment]
This is identical to #618 , which has been recently solved with #718. Here's an example that performs formatting with `fmt_number()` on all numeric columns:

```r
library(gt)

exibble %>%
  gt() %>%
  fmt_number(
    columns = where(is.numeric),
    decimals = 4
  )
```
